pkg/cluster/routing/table.go

```diff
@@ -1,22 +1,41 @@
 package routing
 
 import (
+	"sync"
 	"time"
 
 	"github.com/wetware/casm/pkg/stm"
-	"go.uber.org/atomic"
 )
 
+type Clock struct {
+	time time.Time
+	mu   sync.RWMutex
+}
+
 type Table struct {
-	clock   *atomic.Time
+	clock   *Clock
 	records stm.TableRef
 	sched   stm.Scheduler
 }
 
+func (c *Clock) Load() time.Time {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+	return c.time
+}
+
+func (c *Clock) Store(val time.Time) {
+	c.mu.RLock()
```

Comment:
`Store` is a write operation, so this needs to hold a write-lock (`c.mu.Lock()` and `c.mu.Unlock()`).